### PR TITLE
690 update documentation of params()

### LINKED ISSUE #690: Join documentation of `call_derivation()` and `params()`
These functions can only be used in conjunction so they should be put on the same help page with clear examples to showcase their usage.

### PR DESCRIPTION
**Pull request by traceyxw**
Close #690

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

**Comment by bms63** [comment]
Hi Tracey.  Seeing where we are at with this one as well??  I am happy to review and we can probably ask @bundfussr , @esimms999-gsk  or @kabis-ops  to take a look while Tomas is awy.

**Comment by traceyxw** [comment]
> Hi Tracey. Seeing where we are at with this one as well?? I am happy to review and we can probably ask @bundfussr , @esimms999-gsk or @kabis-ops to take a look while Tomas is awy.

Hi Ben, thanks for checking! Would you please help me review this one?

**Comment by kabis-ops** [comment]
@traceyxw, @bms63 - I can take a look.

File: R/call_derivation.R [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #' @param derivation The derivation function to call
 #' @param variable_params A `list` of arguments that are different across iterations.
```
Comment (kabis-ops) [inline R/call_derivation.R:8]: I would add A `list` of **function** arguments that are different across iterations.

File: R/call_derivation.R [hunk 2]
```diff
@@ -7,7 +7,7 @@
...
 #'   Each set of arguments must be created using [`params()`].
-#' @param ... Any number of *named* arguments that are fixed across iterations.
+#' @param ... Any number of *named* arguments that stay the same across iterations.
```
Comment (kabis-ops) [inline R/call_derivation.R:10]: same here

**Review by kabis-ops — COMMENTED** [review]
Hi @traceyxw, I don't have much comments on current update. 
However I wonder if we should rewrite the examples to make things clearer?
I would start stating that we want to derive both ASTDT and AENDT in ADAE and we could use the function `derive_vars_dt()` for both derivation.
Then I would show the example with the 2 call of derive_vars_dt. And I would highlight what arguments are the same across the 2 derivations, and which are different.
And I would finish with the example call derivation.
I feel that it would show better the purpose of the function.
What do you think? 
or maybe we should create a separate vignette and show other examples how to call this function? @bms63 ?

**Comment by traceyxw** [comment]
> Hi @traceyxw, I don't have much comments on current update. However I wonder if we should rewrite the examples to make things clearer? I would start stating that we want to derive both ASTDT and AENDT in ADAE and we could use the function `derive_vars_dt()` for both derivation. Then I would show the example with the 2 call of derive_vars_dt. And I would highlight what arguments are the same across the 2 derivations, and which are different. And I would finish with the example call derivation. I feel that it would show better the purpose of the function. What do you think? or maybe we should create a separate vignette and show other examples how to call this function? @bms63 ?

Thank you Samia. This is very helpful!

**Comment by bms63** [comment]
I think a separate vignette could be helpful, but we should make a new issue for that.

File: R/call_derivation.R [hunk 3]
```diff
@@ -94,23 +94,73 @@ call_derivation <- function(dataset = NULL, derivation, variable_params, ...) {
...
+#'
+#' ## As shown above, the same values were assigned to arguments `min_dates` and `max_dates`,
+#' ## whereas arguments `new_vars_prefix`, `dtc`, and `date_imputation` were assigned
+#' ## with different values to derive variables `ASTDT` and `AENDT`.
+#'
+#' ## While `derive_vars_dt()` can only add one variable at a time, using `call_derivation()`
+#' ## one can add multiple variables in one go. `params()` is used to create a set of arguments
+#' ## that are different from one another for each variable to be derived and form a `list` of
+#' ## arguments for `variable_params` in `call_derivation()`. The function arguments that stay the
+#' ## same among variables to be derived are specified outside `variable_params`.
```
Comment (kabis-ops) [inline R/call_derivation.R:150]: I would maybe re-phrase?
"While `derive_vars_dt()` can only add one variable at a time, using `call_derivation()` one can add multiple variables in one go. 
The function arguments which are different from a variable to another (e.g.  `new_vars_prefix`, `dtc`, and `date_imputation`) are specified as a `list` of `params()` in the `variable_params` argument of `call_derivation()`.
All other arguments which are common to all variables are specified outside of `variable_params` (i.e. in `...`).

**Review by kabis-ops — COMMENTED** [review]
Hi Tracey, Thanks for the update: just a minor comment about re-phrasing :)

**Review by kabis-ops — APPROVED** [review]

**Pull request merged**